### PR TITLE
PR#283 Created parser for iosxe/show_ap_config_general

### LINKED ISSUE #39: 20. show ap config general
** see chat for details

### PR DESCRIPTION
**Pull request by reustaquio22**

### CODE REVIEW

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        ap_config_general_dict = {}
+
+#         Cisco AP Name   : bg-1-cap1
```
Comment (itdependsnetworks) [inline src/genie/libs/parser/iosxe/show_ap.py]: 1 example is enough, don't want to clutter.

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+            clean_lines = list(filter(None, raw_output.splitlines()))
+            rendered_lines = []
+            for clean_line in clean_lines:
+                clean_line_strip = clean_line.strip()
+                # Remove lines unwanted lines from list of "remove_lines"
+                if not clean_line_strip.startswith(remove_lines):
+                    rendered_lines.append(clean_line_strip)
+            return rendered_lines
+
+        def match_regex(line, regex_patterns):
```
Comment (itdependsnetworks) [inline src/genie/libs/parser/iosxe/show_ap.py]: Why use a different pattern here?

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+
+        out_filter = filter_lines(raw_output=out, remove_lines=remove_lines)
+        ap_name_key = ''
+
+        for line in out_filter:
+            # This output will capture the key values with the ':' as the delimeter
+            captured_groups = match_regex(line=line, regex_patterns=regex_patterns)
+            if captured_groups:
+                # Normalize the key to replace spaces and hyphens with underscores
+                k = captured_groups['key'].strip().replace(' ', '_').replace('-', '_').lower()
```
Comment (itdependsnetworks) [inline src/genie/libs/parser/iosxe/show_ap.py]: Too much magic, imho, would prefer to keep the other pattern. We are not specifying what we are looking for here.

Comment (reustaquio22) [reply src/genie/libs/parser/iosxe/show_ap.py]: I agree that we lose visibility of the pattern that we're trying to match. This sample output has over 100 lines of key value pairs using`:` as the delimiter. The goal was to reduce repetitive coding. Although this class of capturing the raw data is not explicit with what exact line we're matching, the schema is accurate and tested successfully.

Comment (itdependsnetworks) [reply src/genie/libs/parser/iosxe/show_ap.py]: It is tested successfully for the sample data, the issue is that there is no determination on what other data could come. Should be just as easy to generate the data for known use cases and accomplish the same.

File: src/genie/libs/parser/iosxe/show_ap.py (hunk 2 above)
Comment (reustaquio22) [reply src/genie/libs/parser/iosxe/show_ap.py]: This pattern was removed.

File: src/genie/libs/parser/iosxe/show_ap.py (hunk 3 above)
Comment (reustaquio22) [reply src/genie/libs/parser/iosxe/show_ap.py]: Regex was created for every line instead of using the new function.

**Comment by itdependsnetworks** [comment]
Submit a PR please

**Comment by reustaquio22** [comment]
[Cisco PR #216 ](https://github.com/CiscoTestAutomation/genieparser/pull/216)

**Comment by itdependsnetworks** [comment]
This got lost in the shuffle, @qduk, do you mind getting into a new cisco PR? Feel free to just copy the code and create an entirely new branch to avoid merge conflicts.

**Comment by qduk** [comment]
[#283](https://github.com/CiscoTestAutomation/genieparser/pull/283) created for Cisco.

**Comment by itdependsnetworks** [comment]
completed https://github.com/CiscoTestAutomation/genieparser/pull/283

**Pull request closed without merging**